common/handlers/users/patch.go

```diff
@@ -0,0 +1,113 @@
+package users
+
+import (
+	"context"
+
+	dsc "github.com/aserto-dev/go-directory/aserto/directory/common/v3"
+	dsr "github.com/aserto-dev/go-directory/aserto/directory/reader/v3"
+	dsw "github.com/aserto-dev/go-directory/aserto/directory/writer/v3"
+	"github.com/aserto-dev/scim/common"
+	"github.com/aserto-dev/scim/common/convert"
+	"github.com/elimity-com/scim"
+	serrors "github.com/elimity-com/scim/errors"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func (u UsersResourceHandler) Patch(ctx context.Context, id string, operations []scim.PatchOperation) (scim.Resource, error) {
+	logger := u.logger.With().Str("method", "Patch").Str("id", id).Logger()
+	logger.Info().Msg("patch user")
+	logger.Trace().Any("operations", operations).Msg("patching user")
+
+	converter := convert.NewConverter(u.cfg)
+
+	getObjResp, err := u.dirClient.DS().Reader.GetObject(ctx, &dsr.GetObjectRequest{
+		ObjectType:    u.cfg.User.SourceObjectType,
+		ObjectId:      id,
+		WithRelations: false,
+	})
+	if err != nil {
+		logger.Error().Err(err).Msg("failed to get user")
+		st, ok := status.FromError(err)
+
+		if ok && st.Code() == codes.NotFound {
+			return scim.Resource{}, serrors.ScimErrorResourceNotFound(id)
+		}
+
+		return scim.Resource{}, err
+	}
+
+	attr := converter.ObjectToResourceAttributes(getObjResp.GetResult())
+
+	for _, op := range operations {
```

Comment:
I'd extract this entire loop into a function that takes `operations` and returns `(attr, error)`.